src/util/mmap_impl_win32.h

```diff
@@ -0,0 +1,241 @@
+#ifndef JUMANPP_MMAP_IMPL_WIN32_HPP
+#define JUMANPP_MMAP_IMPL_WIN32_HPP
+
+#define NOMINMAX
+#include <windows.h>
+
+namespace jumanpp {
+namespace util {
+
+MappedFile::MappedFile(MappedFile &&o) noexcept
+    : fd_{o.fd_},
+      filename_{std::move(o.filename_)},
+      type_{o.type_},
+      size_{o.size_} {
+  o.fd_ = nullptr;
+}
+
+MappedFile &MappedFile::operator=(MappedFile &&o) noexcept {
+  if (this->fd_ != nullptr) {
+    CloseHandle(this->fd_);
+  }
+  fd_ = o.fd_;
+  filename_ = std::move(o.filename_);
+  type_ = o.type_;
+  size_ = o.size_;
+  o.fd_ = nullptr;
+  return *this;
+}
+
+MappedFile::~MappedFile() {
+  if (this->fd_ != nullptr) {
+      CloseHandle(this->fd_);
+  }
+}
+
+inline static std::wstring to_wide_string(const std::string& str) {
```

Comment:
This function should be probably moved to a more global place, because it would be used on Windows in other filesystem access-related places.
Also, the input parameter should be probably a StringPiece.

Reply:
Ah, I didn't notice that originally file name is passed as StringPiece.
I'll move this function in separate win32 utility unit